### PR TITLE
Update to 0.7

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -32,8 +38,8 @@ install:
 build_script:
 # Need to convert from shallow to complete for Pkg.clone to work
   - IF EXIST .git\shallow (git fetch --unshallow)
-  - C:\projects\julia\bin\julia -e "versioninfo(); Pkg.init();
-      Pkg.clone(pwd(), \"Optimize\"); Pkg.build(\"Optimize\")"
+  - C:\projects\julia\bin\julia -e "VERSION >= v\"0.7\" && using InteractiveUtils, Pkg;
+      versioninfo(); Pkg.init(); Pkg.clone(pwd(), \"Optimize\"); Pkg.build(\"Optimize\")"
```
Comment (dpo) [inline appveyor.yml]: No need for `Pkg.init()` in 0.7 and 1.0. No need for `InteractiveUtils` and `versioninfo()` either. It's printed by default.

**Comment by abelsiqueira** [comment]
Now this is passing. I've rebased to #70, dropped 0.6, and figured how to install Krylov without the release.

**Comment by dpo** [comment]
Looks great, thank you!

File: .travis.yml [hunk 2]
```diff
@@ -24,18 +25,12 @@ notifications:
...
-    export PATH=`pwd`/cmake-3.3.2-Linux-x86_64/bin:$PATH;
-    unset DYLD_LIBRARY_PATH;
-    sudo ln -s /usr/lib/gcc/x86_64-linux-gnu/$(gfortran -dumpversion | cut -f1,2 -d.)/libgfortran.so /usr/local/lib; fi
+  - if [ "$TRAVIS_OS_NAME" == "osx" ]; then brew update && brew cask uninstall --force oclint && brew install gcc; brew tap optimizers/cutest; brew install cutest; brew install mastsif; for f in "archdefs" "sifdecode" "mastsif" "cutest"; do source $(brew --prefix $f)/$f.bashrc; done; fi
+  - if [ "$TRAVIS_OS_NAME" == "linux" ]; then unset DYLD_LIBRARY_PATH; fi
 
 script:
   - if [[ -a .git/shallow ]]; then git fetch --unshallow; fi
-  - julia -e 'Pkg.clone(pwd()); Pkg.build("Optimize"); Pkg.test("Optimize"; coverage=true)'
+  - julia -e 'VERSION >= v"0.7" && using Pkg; Pkg.add(PackageSpec(url="https://github.com/JuliaSmoothOptimizers/Krylov.jl")); Pkg.clone(pwd()); Pkg.build("Optimize"); Pkg.test("Optimize"; coverage=true)'
```
[the remark this answers is not in the document]
Comment (abelsiqueira) [reply .travis.yml]: Fixed

File: examples/run_trunk.jl [hunk 3]
```diff
@@ -1,10 +1,6 @@
 using Optimize
 using NLPModels
 using OptimizationProblems
-using MiniLogging
```
[the remark this answers is not in the document]
Comment (abelsiqueira) [reply examples/run_trunk.jl]: Yes, hadn't touched the examples. Now fixed.

**Comment by dpo** [comment]
Great, thank you!

**Pull request merged**